readme.md

```diff
@@ -8,47 +8,37 @@ Provides a light framework for responding to HTTP requests.
 
 ```php
 <?php
 declare(strict_types=1);
 
-// phpcs:ignoreFile
-
 use corbomite\di\Di;
 use corbomite\http\Kernel as HttpKernel;
 
-define('APP_BASE_PATH', __DIR__);
-define('APP_VENDOR_PATH', APP_BASE_PATH . '/vendor');
-
-require APP_VENDOR_PATH . '/autoload.php';
+require __DIR__ . '/vendor/autoload.php';
 
 Di::get(HttpKernel::class)();
 ```
 
-## CSRF
+### Kernel Method Arguments
 
-Corbomite HTTP implements [CSRF PSR-15 Middleware](https://github.com/Grafikart/PSR15-CsrfMiddleware) to prevent cross-site forgery on post requests. However, sometimes it may be desirable for certain uri starting segments not to implement CSRF protection. In those cases, you can define a constant before the Kernel is run to define the segment or segments that should not implement CSRF protection:
+The Kernel can receive up to two arguments.
 
-```php
-define('CSRF_EXEMPT_SEGMENTS', 'my-segment');
+If the first argument is an array, the second argument will be ignored, and the first argument as an array must send class names or class instances as values of the array that implement `\Psr\Http\Server\MiddlewareInterface`. These classes will be added to the middleware stack after any error handlers are added (and after the CSRF middleware if that's not disabled) and before Corbomite HTTP's Action params and routing.
 
-// or
+If the first argument is a string or a class, it must implement `\Psr\Http\Server\MiddlewareInterface` and will be added as the error handler if the environment variable `DEV_MODE` is not set to a string of `true`. The second argument then can be an array of middleware as above.
 
-define('CSRF_EXEMPT_SEGMENTS', [
-    'my-segment',
-    'another-segment',
-]);
-```
+### Error Handling
 
-## Dev mode
+If the environment variable `DEV_MODE` is set to a string of `true`, the Kernel will attempt to set PHP error output to maximum and add `\Franzl\Middleware\Whoops\WhoopsMiddleware` as the first item to the middleware stack. In this way, as you develop, you can get whatever information and trace you need over your HTTP stack when an error is encountered.
 
-The Kernel looks for an environment variable named `DEV_MODE`. If that is set to a string of `'true'` the Kernel will attempt to crank up PHP error reporting, and will register [Whoops](https://github.com/filp/whoops) for error reporting.
+If not in dev mode, and if you've provided a middleware to handle errors as described above, then your provided error middleware will be the first thing added to the middleware stack. In this way, you have a chance to render error pages in your app in production.
 
-## Error page class
+If you send a string class name, the Kernel will attempt to get the class from the [Corbomite Dependency Injector](https://github.com/buzzingpixel/corbomite-di) and fall back to `new`ing up the class.
```

Comment:
I can't remember if I asked this the other day, but is there anything about this package that specifically requires https://github.com/buzzingpixel/corbomite-di or could it simply require PSR-11's interfaces? Up to you, of course, just curious if there's an opportunity for decoupling here.

Reply:
https://github.com/buzzingpixel/corbomite-http/pull/1/commits/be1fb3507123dd0cbde16a579d5f06795109de79

Reply:
(@kevinsmith  To be clear I left it as a composer dependency for now (one of the other packages relies on it at this time but I removed it as specifically mentioned or used)